Sources/PackageSigning/SignatureProvider.swift

```diff
@@ -232,34 +249,66 @@ struct CMSSignatureProvider: SignatureProviderProtocol {
         }
     }
 
+    private func isValidSignature(
+        signature: [UInt8],
+        content: [UInt8],
+        verifierConfiguration: VerifierConfiguration
+    ) async throws -> CMS.SignatureVerificationResult {
+        var trustRoots: [Certificate] = []
+        if verifierConfiguration.includeDefaultTrustStore {
+            trustRoots.append(contentsOf: CertificateStores.defaultTrustRoots)
+        }
+        trustRoots.append(contentsOf: try verifierConfiguration.trustedRoots.map { try Certificate($0) })
+
+        return await CMS.isValidSignature(
+            dataBytes: content,
+            signatureBytes: signature,
+            // The intermediates supplied here will be combined with those
+            // included in the signature to build cert chain for validation.
+            //
+            // Those who use ADP certs for signing are not required to provide
+            // the entire cert chain, thus we must supply WWDR intermediates
+            // here so that the chain can be constructed during validation.
+            // Whether the signing cert is trusted still depends on whether
+            // the WWDR roots are in the trust store or not, which by default
+            // they are but user may disable that through configuration.
+            additionalIntermediateCertificates: Certificates.wwdrIntermediates,
+            trustRoots: CertificateStore(trustRoots),
+            policy: self.buildPolicySet(configuration: verifierConfiguration, httpClient: self.httpClient)
+        )
+    }
+
+    func extractSigningEntity(
+        signature: [UInt8],
+        verifierConfiguration: VerifierConfiguration
+    ) async throws -> SigningEntity {
```

Comment:
Perhaps this method can just throw/fail immediately and I will implement it as part of https://github.com/apple/swift-package-manager/pull/6316 later when rdar://107475816 is done.

Reply:
Updated to throw.